cmd/livepeer_bench/livepeer_bench.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	//"runtime/pprof"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/lpms/ffmpeg"
+	"github.com/livepeer/m3u8"
+)
+
+func main() {
+	/*
+		cprof, err := os.Create("bench.prof")
+		if err != nil {
+			panic(err)
+		}
+		pprof.StartCPUProfile(cprof)
+		defer pprof.StopCPUProfile()
+	*/
+	// Override the default flag set since there are dependencies that
+	// incorrectly add their own flags (specifically, due to the 'testing'
+	// package being linked)
+	flag.Set("logtostderr", "true")
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fname := flag.String("fname", "", "Input m3u8 manifest file")
+	conc := flag.Int("conc", 1, "# of concurrent transcode sessions")
+	segs := flag.Int("segs", 0, "Maximum # of segments to transcode (default all)")
+	profs := flag.String("profs", "P240p30fps16x9,P360p30fps16x9,P720p30fps16x9", "Transcoding options for broadcast job, or path to json config")
+	nvidia := flag.String("nvidia", "", "Comma-separated list of Nvidia GPU device IDs to use for transcoding")
+	outPrefix := flag.String("outPrefix", "", "Output segments' prefix (no segments are generated by default)")
+
+	flag.Parse()
+
+	if *fname == "" {
+		panic("Please provide the input manifest as `-fname <input.m3u8>`. See -h or -help for more.")
+	}
+
+	profiles := parseVideoProfiles(*profs)
+
+	f, err := os.Open(*fname)
+	if err != nil {
+		panic(err)
+	}
+	p, _, err := m3u8.DecodeFrom(bufio.NewReader(f), true)
+	if err != nil {
+		panic(err)
+	}
+	pl, ok := p.(*m3u8.MediaPlaylist)
+	if !ok {
+		panic("Expecting media PL")
+	}
+
+	accel := ffmpeg.Software
+	devices := []string{}
+	if *nvidia != "" {
+		accel = ffmpeg.Nvidia
+		devices = strings.Split(*nvidia, ",")
+	}
+
+	ffmpeg.InitFFmpeg()
+	segCount := 0
+	var wg sync.WaitGroup
+	dir := path.Dir(*fname)
+	start := time.Now()
+	fmt.Fprintf(os.Stderr, "Program %s Source %s Concurrency %d Profiles %s\n", os.Args[0], *fname, *conc, *profs)
+	fmt.Println("time,stream,segment,length")
```

Comment:
What do you think of using a table format here as well ? I think we could also omit displaying Unix time or is it a requirement for forensics? 

I also added the transcode error here as per the comment below this one but feel free to omit it.

We can initiate the table on this line 

```go

type struct resultsTableData {
 sync.Mutex
 data [][]string
}
resultsTable := tablewriter.NewWriter(os.Stdout)
resultsTable.SetHeader([]string{"Stream No.", "Segment No.", "Latency", "Error"})
resultsData := &resultsTableData{
 data: [][]string{}{}
}
```

Then on L123 we can append to `data` rather than printing out directly 
```go
resultsData.Lock()
resultsData.data = append(resultsData.data, []string{
  fmt.Sprintf("%v", k),
  fmt.Sprintf("%v", j),
  fmt.Sprintf("%v s", time.Since(t).Seconds()),
  err.Error(), 
})
resultsData.Unlock()
```

When the waitgroup finishes, before appending `data` to `resultsTable` we can sort `data` based on 1. Stream No. and 2. Segment No. to get all in order. 

After that we can append the data to the table and render it

```go
resultsData.Lock()
for _, v := range resultsData.data {
    table.Append(v)
}
resultsData.Unlock()
table.SetAlignment(tablewriter.ALIGN_RIGHT)
table.SetCenterSeparator("*")
table.SetRowLine(true)
table.SetColumnSeparator("|")
table.Render()
```

Reply:
Yeah this would be neat, and separate out diff streams, will do.

I'd still prefer keeping the data printing in a csv/tsv like format though - as one of the requirements in the spec #1664 was to let the user export data for more processing with other tools. And almost everyone could export a tsv into excel/sheets but `|` separator might give trouble.